pkgs/development/misc/juce/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+
+# Native build inputs
+, cmake
+, pkg-config
+, makeWrapper
+
+# Dependencies
+, alsa-lib
+, freetype
+, curl
+, libglvnd
+, webkitgtk
+, pcre
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "juce";
+  version = "7.0.7";
+
+  src = fetchFromGitHub {
+    owner = "juce-framework";
+    repo = "juce";
+    rev = finalAttrs.version;
+    hash = "sha256-r+Wf/skPDexm3rsrVBoWrygKvV9HGlCQd7r0iHr9avM=";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
+
+  buildInputs = [
+    freetype # libfreetype.so
+    curl # libcurl.so
+    alsa-lib # libasound.so
+    libglvnd # libGL.so
+    stdenv.cc.cc.lib # libstdc++.so libgcc_s.so
+    webkitgtk # webkit2gtk-4.0
+    pcre # libpcre2.pc
+  ];
+
+  meta = with lib; {
+    description = "JUCE is an open-source cross-platform C++ application framework for desktop and mobile applications, including VST, VST3, AU, AUv3, RTAS and AAX audio plug-ins";
+    homepage = "https://github.com/juce-framework/JUCE";
+    license = with licenses; [ isc gpl3 ];
```

Comment:
`gpl3` is deprecated, it should be either `gpl3Only` or `gpl3Plus`. I couldn't find out which one JUCE is using though, so you should consider opening an issue to them asking them which one they're using

Reply:
Likely `gpl3Plus`. Upstream puts all their core modules under their ISC license and state all other modules are covered by GPL commercial license.